balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -456,30 +473,73 @@ func (b *pickfirstBalancer) requestConnectionLocked() {
 		switch scd.state {
 		case connectivity.Idle:
 			scd.subConn.Connect()
+			b.scheduleNextConnectionLocked()
+			return
 		case connectivity.TransientFailure:
-			// Try the next address.
+			// The SubConn is being re-used and failed during a previous pass
+			// over the addressList. It has not completed backoff yet.
+			// Mark it as having failed and try the next address.
+			scd.connectionFailed = true
```

Comment:
`connectionFailed` is a bit like `lastErr != nil`.  Do we need both?

Reply:
OK, I see what's happening here, thanks for the explanation.

Maybe name it `connectionFailed(In/During)FirstPass`?

Reply:
Renamed to `connectionFailedInFirstPass`.